src/playback-watcher.js

```diff
@@ -211,6 +211,24 @@ export default class PlaybackWatcher {
       this.timer_ = setTimeout(this.skipTheGap_.bind(this),
                                difference * 1000,
                                currentTime);
+      return;
+    }
+
+    let currentRange = Ranges.findRange(buffered, currentTime);
+
+    // Sometimes the player can stall for unknown reasons within a contiguous buffered
+    // region with no indication that anything is amiss (seen in Firefox). Seeking to
+    // currentTime is usually enough to kickstart the player.
+    if (currentRange.length) {
```

Comment:
That was why I originally had the 1 second safety net. The `Ranges.findRange` function uses `TIME_FUDGE_FACTOR` which I thought was more inline with your previous comment. I'll make sure to test this exact scenario and see if it triggers the event, and if so, we can discuss an appropriate fudge factor to protect against this